WalletWasabi/Crypto/CryptoGuards.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using WalletWasabi.Crypto.Groups;
+
+namespace WalletWasabi.Crypto
```

Comment:
Can you move this into the Guard instead? The Guard is a Wasabi specific guard, it wasn't intended to be a .NET generic guard.

Reply:
Ok, I didn't do it because I can be free to touch in an isolated way. I will not forget to move it but I will merge it as it is now.